### PR TITLE
ci: Update "Update flake lock" action.

### PR DESCRIPTION
**Pull request by drupol**
This PR:

- [x] Is a follow-up of #54

### CODE REVIEW

File: .github/workflows/upgrade-flakes.yaml [hunk 1]
```diff
@@ -1,10 +1,22 @@
...
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+        with:
+            fetch-depth: 1
+
+      - name: Install Nix
+        uses: cachix/install-nix-action@v16
+        with:
+          nix_path: nixpkgs=channel:nixos-unstable
```
Comment (jtojnar) [inline .github/workflows/upgrade-flakes.yaml:19]: Do we need this line?

File: .github/workflows/upgrade-flakes.yaml [hunk 2]
```diff
@@ -1,10 +1,22 @@
...
+
+      - name: Update flake.lock
+        uses: DeterminateSystems/update-flake-lock@v6
```
Comment (jtojnar) [inline .github/workflows/upgrade-flakes.yaml:22]: Final new line is missing.

File: .github/workflows/upgrade-flakes.yaml (hunk 1 above)
Comment (drupol) [reply .github/workflows/upgrade-flakes.yaml]: I removed it.

File: .github/workflows/upgrade-flakes.yaml (hunk 2 above)
Comment (drupol) [reply .github/workflows/upgrade-flakes.yaml]: Fixed.

**Comment by jtojnar** [comment]
Could you please squash the first two commits. And also add an action for checking editorconfig if we are to have it?

**Comment by drupol** [comment]
> And also add an action for checking editorconfig if we are to have it?

Why would we want to check `.editorconfig` with Github actions?

**Comment by jtojnar** [comment]
Sure it can, that’s how most packages are updated.

**Comment by drupol** [comment]
> Sure it can, that’s how most packages are updated.

I still do not understand the point of checking for such a file, in Github action :)

**Comment by jtojnar** [comment]
I meant checking files in the repository for compliance with the `.editorconfig` since otherwise people without editorconfig plug-in installed will start breaking it unknowingly.

**Comment by drupol** [comment]
@jtojnar I finally did it.

I didn't use `eclint` because it's unmaintained. I used the most up-to-date editorconfig checker: https://github.com/editorconfig-checker/editorconfig-checker which seems to be well maintained.

Let me know if it's ok for you now.

File: .ecrc [hunk 3]
```diff
@@ -0,0 +1,5 @@
+{
```
Comment (jtojnar) [inline .ecrc:1]: Is this file needed?

File: .github/workflows/editorconfig-checks.yaml [hunk 4]
```diff
@@ -0,0 +1,16 @@
+name: Editorconfig checks
```
Comment (jtojnar) [inline .github/workflows/editorconfig-checks.yaml:1]: EditorConfig is spelt with capital C.

File: .github/workflows/upgrade-flakes.yaml [hunk 5]
```diff
@@ -1,10 +1,20 @@
-name: 'Update flakes'
-
+name: update-flake-lock
```
Comment (jtojnar) [inline .github/workflows/upgrade-flakes.yaml:1]: The name should be human readable.

File: .github/workflows/editorconfig-checks.yaml [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+jobs:
+  lockfile:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 1
+
+      - name: Editorconfig checks
```
Comment (jtojnar) [inline .github/workflows/editorconfig-checks.yaml:15]: This should be an action, e.g. “Check files for EditorConfig compliance”

**Review by jtojnar — COMMENTED** [review]
Few more nits.

File: .ecrc (hunk 3 above)
Comment (drupol) [reply .ecrc:1]: I added it myself in my local to prevent the `node_modules` directory to be added to the files to lint.
Maybe on GH it's not necessary, going to try.

File: .github/workflows/editorconfig-checks.yaml (hunk 4 above)
Comment (drupol) [reply .github/workflows/editorconfig-checks.yaml]: Fixed

File: .github/workflows/upgrade-flakes.yaml (hunk 5 above)
Comment (drupol) [reply .github/workflows/upgrade-flakes.yaml]: Fixed

File: .github/workflows/editorconfig-checks.yaml (hunk 6 above)
Comment (drupol) [reply .github/workflows/editorconfig-checks.yaml]: Fixed.

File: .ecrc (hunk 3 above)
Comment (drupol) [reply .ecrc]: Indeed, that file is not necessary on GH because it uses a docker image and the `node_modules` directory doesn't pollute the repository.

Comment (drupol) [reply .ecrc]: I removed it.

**Review by jtojnar — APPROVED** [review]
LGTM, thanks.
